src/upnp_xml.cc

```diff
@@ -845,10 +846,12 @@ void UpnpXMLBuilder::addResources(const std::shared_ptr<CdsItem>& item, pugi::xm
             continue;
         }
 
+        // TODO FIXME Should these be attributes or something else?
+        std::map<std::string, std::string> subtitleQuirks;
         if (handlerType == CH_DEFAULT && !captionInfoEx.empty() && quirks && quirks->checkFlags(QUIRK_FLAG_PV_SUBTITLES)) {
             auto captionInfo = captionInfoEx[0];
-            resAttrs["pv:subtitleFileType"] = toUpper(captionInfo["sec:type"]);
-            resAttrs["pv:subtitleFileUri"] = captionInfo[""];
+            subtitleQuirks["pv:subtitleFileType"] = toUpper(captionInfo["sec:type"]);
+            subtitleQuirks["pv:subtitleFileUri"] = captionInfo[""];
```

Comment:
You need to pass the quirks map to `renderResource` for output as xml attributes. Currently this is dead code.

Other option is to return `xml_node` from `renderResource`

Reply:
Yep, too hasty on the push